src/gamepad/gamepad-monitor.vala

```diff
@@ -0,0 +1,59 @@
+// This file is part of GNOME Games. License: GPLv3
+
+/**
+ * This class provides a way to the client to monitor gamepads
+ *
+ * The client interfaces with this class primarily
+ */
+public class LibGamepad.GamepadMonitor : Object {
+	/**
+	 * Emitted when a gamepad is plugged in
+	 * @param  gamepad    The gamepad
+	 */
+	public signal void gamepad_plugged (Gamepad gamepad);
+
+	public delegate void GamepadCallback (Gamepad gamepad);
+
+	private static GamepadMonitor instance;
+
+	private SList <Gamepad> gamepads;
+	private RawGamepadMonitor raw_gamepad_monitor;
```

Comment:
As you made it a singleton you don't need to store it anymore: the instance will never be destroyed.